terraform-aws-static-site/main.tf

```
@@ -0,0 +1,163 @@
+# Non www-bucket
+resource "aws_s3_bucket" "b" {
+  bucket = var.bucket_name
+  acl    = "public-read"
+
+  website {
+    index_document = "index.html"
+    error_document = "index.html"
+  }
+}
+
+data "aws_iam_policy_document" "b" {
+  statement {
+    sid    = "PublicReadGetObject"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    actions   = ["s3:GetObject"]
+    resources = ["arn:aws:s3:::${var.bucket_name}/*"]
+  }
+}
+
+resource "aws_s3_bucket_policy" "b" {
+  bucket = aws_s3_bucket.b.id
+  policy = data.aws_iam_policy_document.b.json
+}
+
+# www-bucket
+resource "aws_s3_bucket" "www-bucket" {
+  bucket = "www-${var.bucket_name}"
+  acl    = "public-read"
+
+  website {
+    redirect_all_requests_to = "https://${var.redirect_address}"
+  }
+}
+
+# Non www-bucket CDN
+resource "aws_cloudfront_distribution" "s3_distribution" {
+  origin {
+    domain_name = aws_s3_bucket.b.bucket_regional_domain_name
+    origin_id   = var.origin_id
+
```

Additionally, we need to add an s3_origin_config block

```suggestion

    s3_origin_config {
      origin_access_identity = aws_cloudfront_origin_access_identity.origin_access_identity.cloudfront_access_identity_path
    }
```